rust/src/pgsql/log.rs

```
@@ -0,0 +1,420 @@
+            PgsqlErrorNoticeFieldType::Position => {
+                jb.set_string_from_bytes("position", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::InternalPosition => {
+                jb.set_string_from_bytes("internal_position", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::InternalQuery => {
+                jb.set_string_from_bytes("internal_query", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::Where => {
+                jb.set_string_from_bytes("where", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::SchemaName => {
+                jb.set_string_from_bytes("schema_name", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::TableName => {
+                jb.set_string_from_bytes("table_name", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::ColumnName => {
+                jb.set_string_from_bytes("column_name", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::DataType => {
+                jb.set_string_from_bytes("data_type", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::ConstraintName => {
+                jb.set_string_from_bytes("constraint_name", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::File => {
+                jb.set_string_from_bytes("file", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::Line => {
+                jb.set_string_from_bytes("line", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::Routine => {
+                jb.set_string_from_bytes("routine", &field.field_value)?;
+            },
+            PgsqlErrorNoticeFieldType::TerminatorToken => {
+                // do nothing, as that's the terminator token ?
+            },
+            PgsqlErrorNoticeFieldType::UnknownFieldType => {
```

is this unknown to suri, or unknown in the pgsql protocol spec?

---

The specification lists existing field types, but states that more types can be added in the future - therefore, frontends should silently ignore unrecognized field types.